data/signUp.js

```diff
@@ -1,24 +1,53 @@
-import {ObjectId} from 'mongodb';
-import {user} from '../config/mongoCollections.js';
-import validations from '../helpers.js';
-const userCollection = await user();
+import { ObjectId } from "mongodb";
+import { users } from "../config/mongoCollections.js";
+import validations from "../helpers.js";
+const userCollection = await users();
 
 const signUpFunctions = {
-
-    async create(fname, lname, age, email, password) {
-
-        age = Number(age)
-
-        if (!validations.isProperString([fname, lname, email, password])) throw "Error : FirstName, Last Name, Email, Password can only be string not just string with empty spaces"
-        validations.isAge(age)
-
-        const ifAlready = await userCollection.findOne({email: email})
-        if (ifAlready) throw "Error: User Email is already registered"
-
-        const finalPush = await userCollection.insertOne({fname, lname, email, password, profile:[{age}]})
-        return (await userCollection.findOne({_id: finalPush.insertedId}))
-    }
-
-}
-
-export default signUpFunctions
\ No newline at end of file
+  async create(fname, lname, age, email, password) {
+    age = Number(age);
+
+    console.log("got here");
+    if (!validations.isProperString([fname, lname, email, password]))
+      throw "Error : FirstName, Last Name, Email, Password can only be string not just string with empty spaces";
+    validations.isAge(age);
+
+    const ifAlready = await userCollection.findOne({ email: email });
+    if (ifAlready) throw "Error: User Email is already registered";
+
+    // attributes need, but to be populated later when profile filled out by user
+    let gender = "";
+    let locationState = "";
+    let university = "";
+    let collegeMajor = "";
+    let interestArea = [];
+    let experience = 0;
+    let seekingJob = [];
+    let connections = [];
+    let group = [];
+    let createdAt = "";
+    let updatedAt = "";
+
+    const finalPush = await userCollection.insertOne({
+      fname,
+      lname,
+      email,
+      password,
+      profile: [{ age }],
```

Comment:
Would it be like age : age? Or, any thoughts on this part?

Reply:
I agree with a validation file, I can start writing that, and refactor current existing error checking into that file.
This is definitely helpful to manage this in a good way. Will push up in my next branch.

for age, I would need to look further into this - I think this was a commit from Pradyumn. However, I agree that just having age: age would be simpler and easier here